From 5a01974f788648afcf28179abbe6b41761a10539 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Mon, 9 Nov 2015 11:28:16 +0800
Subject: [PATCH 0660/5242] MLK-11838 ARM: dts: imx6sx: fix suspend/resume
 failed on i.mx6sx-sdb-reva board

commit  ed79aa2cbf17bd893113cbd957f1f5504dbc7738 from
https://source.codeaurora.org/external/imx/linux-imx.git

There is one ROM bug on i.mx6sx TO1.1 that default ROM code use the old
ocram space 0x900000, but it should be 0x8f8000. So SW workaround need to
be added, unfortunately, the 'romcp' device node not added in i.mx6sx dtsi
on v4.1, thus i.mx6sx-sdb-reva board will be failed to resume back, add them
back.

Signed-off-by: Robin Gong <b38343@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index a8aa7d5..383d301 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -1141,6 +1141,11 @@
 				};
 			};
 
+			romcp@21ac000 {
+					compatible = "fsl,imx6sx-romcp", "syscon";
+					reg = <0x021ac000 0x4000>;
+			};
+
 			sai1: sai@21d4000 {
 				compatible = "fsl,imx6sx-sai";
 				reg = <0x021d4000 0x4000>;
-- 
1.7.9.5

